src/test/java/coydir/logic/commands/ViewDepartmentCommandTest.java

```diff
@@ -0,0 +1,167 @@
+                ViewDepartmentCommand.MESSAGE_VIEW_DEPARTMENT_SUCCESS, "Operations");
+        assertViewDepartmentCommandSuccess(viewDepartmentCommand, model, expectedMessage,
+                expectedModel, "Operations");
+    }
+
+    @Test
+    public void viewProductManagementDepartmentSuccess() {
+        ViewDepartmentCommand viewDepartmentCommand = new ViewDepartmentCommand("Product Management");
+        String expectedMessage = String.format(
+                ViewDepartmentCommand.MESSAGE_VIEW_DEPARTMENT_SUCCESS, "Product Management");
+        assertViewDepartmentCommandSuccess(viewDepartmentCommand, model, expectedMessage,
+                expectedModel, "Product Management");
+    }
+
+    @Test
+    public void viewProductionDepartmentSuccess() {
+        ViewDepartmentCommand viewDepartmentCommand = new ViewDepartmentCommand("Production");
+        String expectedMessage = String.format(
+                ViewDepartmentCommand.MESSAGE_VIEW_DEPARTMENT_SUCCESS, "Production");
+        assertViewDepartmentCommandSuccess(viewDepartmentCommand, model, expectedMessage,
+                expectedModel, "Production");
+    }
+
+    @Test
+    public void viewQualityAssuranceDepartmentSuccess() {
+        ViewDepartmentCommand viewDepartmentCommand = new ViewDepartmentCommand("Quality Assurance");
+        String expectedMessage = String.format(
+                ViewDepartmentCommand.MESSAGE_VIEW_DEPARTMENT_SUCCESS, "Quality Assurance");
+        assertViewDepartmentCommandSuccess(viewDepartmentCommand, model, expectedMessage,
+                expectedModel, "Quality Assurance");
+    }
+
+    @Test
+    public void viewResearchAndDevelopmentDepartmentSuccess() {
+        ViewDepartmentCommand viewDepartmentCommand = new ViewDepartmentCommand("Research and Development");
+        String expectedMessage = String.format(
+                ViewDepartmentCommand.MESSAGE_VIEW_DEPARTMENT_SUCCESS, "Research and Development");
+        assertViewDepartmentCommandSuccess(viewDepartmentCommand, model, expectedMessage,
+                expectedModel, "Research and Development");
+    }
+
+    @Test
+    public void viewSalesDepartmentSuccess() {
+        ViewDepartmentCommand viewDepartmentCommand = new ViewDepartmentCommand("Sales");
+        String expectedMessage = String.format(
+                ViewDepartmentCommand.MESSAGE_VIEW_DEPARTMENT_SUCCESS, "Sales");
+        assertViewDepartmentCommandSuccess(viewDepartmentCommand, model, expectedMessage,
+                expectedModel, "Sales");
+    }
+
+    @Test
+    public void viewTechnologyDepartmentSuccess() {
+        ViewDepartmentCommand viewDepartmentCommand = new ViewDepartmentCommand("Technology");
+        String expectedMessage = String.format(
+                ViewDepartmentCommand.MESSAGE_VIEW_DEPARTMENT_SUCCESS, "Technology");
+        assertViewDepartmentCommandSuccess(viewDepartmentCommand, model, expectedMessage,
+                expectedModel, "Technology");
+    }
+}
```

Comment:
I think similar to what Kevin mentioned. Perhaps can add more failure cases when the user messes up the view command, like view non-existent department or something